### PR TITLE
QuickLogic: upstream SDC, PCF and post-synthesis fixes

### PR DESCRIPTION
**Pull request by acomodi**
This PR upstreams some feature enhancements around SDC, PCF parsing and post-synth simulation fixes.

It also adds and toolchain installation test.

### CODE REVIEW

File: quicklogic/qlf_k4n8/tests/design_flow/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,6 @@ add_subdirectory(and2_latch)
 add_subdirectory(bin2bcd)
 add_subdirectory(clock_tree_design)
 add_subdirectory(full_adder)
-add_subdirectory(io_reg)
```
Comment (mkurc-ant) [inline quicklogic/qlf_k4n8/tests/design_flow/CMakeLists.txt:12]: Any reason why this one is removed?

**Review by mkurc-ant — APPROVED** [review]
One comment, otherwise LGTM as long as CI passes.

File: quicklogic/qlf_k4n8/tests/design_flow/CMakeLists.txt (hunk 1 above)
Comment (acomodi) [reply quicklogic/qlf_k4n8/tests/design_flow/CMakeLists.txt:12]: Possibly, this is a rebase error, I'll add this one back

Comment (acomodi) [reply quicklogic/qlf_k4n8/tests/design_flow/CMakeLists.txt:12]: Actually, the `io_reg` test is tested in the `features` test, and this was actually removed downstream as well.

**Pull request merged**